File: Array/1ReverseArray/1UsingExtraArray/solution.java
public static int[] reverseArrayExtraArray (int[] arr)
{
  // Storing the size of the original array to be used later at multiple places
  int size = arr.length;

  // Defining a new array
  int[] reversedArr = new int[size];

  // Populating the new array with the elements of the original array in the reversed order
  for (int i = 0; i < size; i++)
    reversedArr[i] = arr[size - i - 1];
  
  return reversedArr;
}

File: Array/1ReverseArray/1UsingExtraArray/solution.c
int* reverse_array_extra_array(int arr[], int size)
{
  // Defining a new array
  int* reversed_arr = malloc(size * sizeof(int));
  // Handling memory allocation failure for the new array
  if (reversed_arr == NULL)
    return NULL;

  // Populating the new array with the elements of the original array in the reverse order
  for (int i = 0; i < size; i++)
    reversed_arr[i] = arr[size - i - 1];
  
  return reversed_arr;
}
